dialer.go

```diff
@@ -195,17 +209,20 @@ func (d *Dialer) Dial(ctx context.Context, instance string, opts ...DialOption)
 		endInfo(err)
 		return nil, err
 	}
-	addr, tlsCfg, err := i.ConnectInfo(ctx, cfg.ipType)
+	ci, err := i.ConnectInfo(ctx, cfg.ipType)
 	if err != nil {
 		endInfo(err)
 		return nil, err
 	}
 	endInfo(err)
+	if cErr := verifyIAMAuthNConfig(ci.Version, cfg.refreshCfg.UseIAMAuthN); cErr != nil {
```

Comment:
Any reason not to inline this check?

Reply:
My only reason is that Dial is getting long and this function adds just enough branches that pulling it out helps readability.